File: app/src/main/java/com/futabooo/android/booklife/screen/booklist/BookListService.java
package com.futabooo.android.booklife.screen.booklist;

import com.google.gson.JsonObject;
import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface BookListService {

  @GET("/users/{user_id}/books/{book_list_menu}") Observable<ResponseBody> get(
      @Path("user_id") int userId,
      @Path("book_list_menu") String bookListMenu);

  @GET("/users/{user_id}/books/{book_list_menu}.json") Observable<JsonObject> getJson(
      @Header("X-CSRF-Token") String csrfToken,
      @Path("user_id") int userId,
      @Path("book_list_menu") String bookListMenu,
      @Query("attach_review") String attach_review,
      @Query("offset") int offset,
      @Query("limit") int limit);
}

File: app/src/main/kotlin/com/futabooo/android/booklife/screen/booklist/BookListService.kt
package com.futabooo.android.booklife.screen.booklist

import com.google.gson.JsonObject
import io.reactivex.Observable
import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Path
import retrofit2.http.Query

interface BookListService {

  @GET("/users/{user_id}/books/{book_list_menu}") fun get(
      @Path("user_id") userId: Int,
      @Path("book_list_menu") bookListMenu: String): Observable<ResponseBody>

  @GET("/users/{user_id}/books/{book_list_menu}.json") fun getJson(
      @Header("X-CSRF-Token") csrfToken: String,
      @Path("user_id") userId: Int,
      @Path("book_list_menu") bookListMenu: String,
      @Query("attach_review") attach_review: String,
      @Query("offset") offset: Int,
      @Query("limit") limit: Int): Observable<JsonObject>
}
